app/ios/Podfile

```diff
@@ -36,5 +36,21 @@ end
 post_install do |installer|
   installer.pods_project.targets.each do |target|
     flutter_additional_ios_build_settings(target)
+    target.build_configurations.each do |config|
+      # You can remove unused permissions here
+      # for more information: https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler_apple/ios/Classes/PermissionHandlerEnums.h
+      # e.g. when you don't need camera permission, just add 'PERMISSION_CAMERA=0'
+      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
+        '$(inherited)',
+      ## dart: PermissionGroup.calendar
+      'PERMISSION_EVENTS=1',
+
+       ## dart: PermissionGroup.calendarFullAccess
+       'PERMISSION_EVENTS_FULL_ACCESS=1',
+ 
+      ## dart: PermissionGroup.notification
+      'PERMISSION_NOTIFICATIONS=1',
+      ]
+    end
```

Comment:
Is this the actual fix then? I don't understand why this would fix anything. Clearly notifications worked before without this, so whatever this is doing, it isn't overwriting the `Info.plist` settings. And as said before, the same is true for calendar events - it is used to work fine before, why is this necessary now?

Reply:
there is a huge explanation but I still don't know where this code comes from. It is not on the [`permission_handler`](https://pub.dev/packages/permission_handler) setup README as implied. If this is not the setup usually needed for permission_handler, why do we need it?